server/jobs/reruns.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable max-len */
+/* eslint-disable no-console */
+import dotenv from 'dotenv';
+import db from '../models';
+
+import { getOrCreateProject, assignProject } from '../modules/freckle/projects';
+import { accessChannel } from '../modules/slack/slackIntegration';
+import {
+  sendDevOnboardingMail, sendSOPOnboardingMail, sendITOffboardingMail, sendSOPOffboardingMail,
+} from '../modules/email/emailModule';
+import { executeEmailAutomation } from './helpers';
+import { findPartnerById } from '../modules/allocations';
+import { retryAutomations } from '../modules/automations';
+
+dotenv.config();
+const { SLACK_AVAILABLE_DEVS_CHANNEL_ID, SLACK_RACK_CITY_CHANNEL_ID } = process.env;
+
+const {
+  SlackAutomation, FreckleAutomation,
+} = db;
+
+const retryAccessChannels = async (slackAutomation, status, existingPlacement, automationId, channelId, context) => {
+  if (slackAutomation.status === status && slackAutomation.type === context) {
+    const response = await accessChannel(existingPlacement.email, channelId, context);
+    retryAutomations(SlackAutomation, { ...response, automationId }, slackAutomation);
+  }
+};
+
+export const freckleAutomationsReruns = (freckleAutomations, automationId) => {
+  freckleAutomations.forEach((freckleAutomation) => {
+    if (freckleAutomation.status === 'failure' && freckleAutomation.type === 'projectCreation') {
+      getOrCreateProject(freckleAutomation.partnerName).then(async (project) => {
+        retryAutomations(FreckleAutomation, { ...project, automationId }, freckleAutomation);
+      });
+    }
+    if (freckleAutomation.status === 'failure' && freckleAutomation.type === 'projectAssignment') {
```

Comment:
We could save the processor from checking if `freckleAutomation.status === 'failure'` twice by moving the content of this if statement into the first in other to have something like:
```
const types = {
  projectCreation: () =>
    getOrCreateProject(freckleAutomation.partnerName).then(async project => {
      retryAutomations(
        FreckleAutomation,
        { ...project, automationId },
        freckleAutomation
      );
    }),
  projectAssignment: () =>
    getOrCreateProject(freckleAutomation.partnerName).then(async project => {
      assignProject(freckleAutomation.email, project.id).then(result => {
        retryAutomations(
          FreckleAutomation,
          { ...result, automationId },
          freckleAutomation
        );
      });
    })
};
if (freckleAutomation.status === "failure") {
  types[freckleAutomation.type]();
}
```

Reply:
resolved